### PR TITLE
docs($http): update YQL currency exchange API example

### LINKED ISSUE #16130: outdated example with Yahoo Finance API
**I'm submitting a ...**

- [x] bug report
- [ ] feature request
- [ ] other (Please do not submit support requests here (see above))

**Current behavior:**

In the Conceptual Overview Section > Accessing the backend> in finance3.js the example with Yahoo Finance API is outdated. Currently the Yahoo Finance API is broken.

**Expected / new behavior:**

In the section Conceptual Overview> Accessing the backend > finance3.js:
I suggest adding a new example that is not outdated or soon to be outdated. This will keep the documentation up to date and increase its readability.

**Minimal reproduction of the problem with instructions:**

**Angular version:** 1.x.y

**Browser:** [all | Chrome XX | Firefox XX | Edge XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

**Anything else:**

One example that could replace the Yahoo Finance  API is the API made by https://docs.openexchangerates.org/ 

Other suggestions:
https://1forge.com/forex-data-api
http://www.financialcontent.com/support/documentation/json_quote_api.php

### PR DESCRIPTION
**Pull request by andypotts**
Closes #16130

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
docs update

**What is the current behavior? (You can also link to an open issue here)**
Currently, the example uses YQL to fetch currency exchange rates (#16130)

**What is the new behavior (if this is a feature change)?**
I have updated the example to a simpler solution that uses the Fixer.io API.  

**Does this PR introduce a breaking change?**
No

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Other information**:

### CODE REVIEW

File: docs/content/guide/concepts.ngdoc [hunk 1]
```diff
@@ -186,7 +186,7 @@ Right now, the `InvoiceController` contains all logic of our example. When the a
 is a good practice to move view-independent logic from the controller into a
 <a name="service">{@link services service}</a>, so it can be reused by other parts
 of the application as well. Later on, we could also change that service to load the exchange rates
-from the web, e.g. by calling the Yahoo Finance API, without changing the controller.
+from the web, e.g. by calling the Fixer.io exchange rate API, without changing the controller.
```
Comment (gkalpak) [inline docs/content/guide/concepts.ngdoc]: Make `Fixer.io` a link.

File: docs/content/guide/concepts.ngdoc [hunk 2]
```diff
@@ -300,7 +300,7 @@ to something shorter like `a`.
 
 ## Accessing the backend
 
-Let's finish our example by fetching the exchange rates from the Yahoo Finance API.
+Let's finish our example by fetching the exchange rates from the Fixer.io exchange rate API.
```
Comment (gkalpak) [inline docs/content/guide/concepts.ngdoc]: Make `Fixer.io` a link.

File: docs/content/guide/security.ngdoc [hunk 3]
```diff
@@ -100,7 +100,7 @@ Protection from JSON Hijacking is provided if the server prefixes all JSON reque
 AngularJS will automatically strip the prefix before processing it as JSON.
 For more information please visit {@link $http#json-vulnerability-protection JSON Hijacking Protection}.
 
-Bear in mind that calling `$http.jsonp`, like in [our Yahoo! finance example](https://docs.angularjs.org/guide/concepts#accessing-the-backend),
+Bear in mind that calling `$http.jsonp`, like in [our currency exchange example](https://docs.angularjs.org/guide/concepts#accessing-the-backend),
```
Comment (gkalpak) [inline docs/content/guide/security.ngdoc]: Actually, this `line in...` clause is obsolete. We no longer use `jsonp` in that example for quite some time.
Please remove that part.

**Review by gkalpak — CHANGES_REQUESTED** [review]
A couple of nits. LGTM otherwise.
:100: for simplifying the example.

**Comment by gkalpak** [comment]
You know what would be awesome (not necessarily in this PR)?
Adding some end-to-end tests for that example :grin:

**Comment by andypotts** [comment]
@gkalpak I've fixed those issues, and agree that some end-to-end tests would be good. When I get a chance I'll create another PR 😄

File: docs/content/guide/concepts.ngdoc [hunk 4]
```diff
@@ -186,7 +186,7 @@ Right now, the `InvoiceController` contains all logic of our example. When the a
 is a good practice to move view-independent logic from the controller into a
 <a name="service">{@link services service}</a>, so it can be reused by other parts
 of the application as well. Later on, we could also change that service to load the exchange rates
-from the web, e.g. by calling the Yahoo Finance API, without changing the controller.
+from the web, e.g. by calling the <a href="http://fixer.io" title="fixer" target="_blank">Fixer.io</a> exchange rate API, without changing the controller.
```
Comment (gkalpak) [inline docs/content/guide/concepts.ngdoc]: `.ngdoc` files understand markdown. You can use `[Fixer.io](http://fixer.io)` for brevity.

File: docs/content/guide/security.ngdoc [hunk 5]
```diff
@@ -100,7 +100,7 @@ Protection from JSON Hijacking is provided if the server prefixes all JSON reque
 AngularJS will automatically strip the prefix before processing it as JSON.
 For more information please visit {@link $http#json-vulnerability-protection JSON Hijacking Protection}.
 
-Bear in mind that calling `$http.jsonp`, like in [our Yahoo! finance example](https://docs.angularjs.org/guide/concepts#accessing-the-backend),
+Bear in mind that calling `$http.jsonp`,
```
Comment (gkalpak) [inline docs/content/guide/security.ngdoc]: Remove the `,`. It is not necessary any more.

**Review by gkalpak — CHANGES_REQUESTED** [review]

**Comment by andypotts** [comment]
@gkalpak thanks, hopefully it's good to go now!

**Review by gkalpak — APPROVED** [review]

**Pull request closed without merging**